app/routes/groups.tsx

```diff
@@ -1,15 +1,67 @@
-import { useGroups } from '~/hooks/useGroups';
-import { Link } from '@remix-run/react';
+import { Form, Link, useLoaderData } from '@remix-run/react';
 import { Image } from '~/components/ui/images';
 import { LinkButton } from '~/components/ui/forms';
+import { db } from '~/modules/database/db.server';
+import { type LoaderFunctionArgs } from '@remix-run/node';
+import type { Group } from '@prisma/client';
+
+export async function loader({ request }: LoaderFunctionArgs) {
+  const url = new URL(request.url);
+  const query = url.searchParams.get('q');
+
+  if (!query) {
+    const groups = await db.group.findMany({ take: 24 });
+    return { groups };
+  }
+
+  const groups = await db.group.findMany({
+    where: { name: { contains: query } },
+  });
+  return { groups };
+}
 
 export default function Group() {
-  const groups = useGroups();
+  const { groups } = useLoaderData<typeof loader>();
+  console.log('groups', groups);
   return (
     <div className="bg-primary">
       <div className="m-auto max-w-screen-xl">
+        <Form method="GET" action="/groups" className="w-full bg-gray-800">
+          <div className="flex p-2 sm:p-8 w-full mx-auto max-w-screen-2xl">
+            <input
+              type="search"
+              aria-label="Search For Events"
+              id="q"
+              name="q"
+              className="block w-full py-4 pl-10 text-xs sm:text-base text-gray-900 md:text-center lg:text-2xl border-none rounded-l-lg italic bg-gray-50 focus:ring-2 focus:outline-none focus:ring-blue-500 text-center"
+              placeholder="Search For Groups"
+              minLength={2}
+              maxLength={50}
```

Comment:
Are there any significance to these limits? Just curious, why did we chose 2 and 50?

Reply:
We could make those constant values that we share across our search fields maybe?